python/rbac/ldap.py

```diff
@@ -143,25 +168,54 @@ def authenticate_ldap_user(self, ldap_user, password):
     def get_user_model(self):
         return User
 
+    @contextmanager
+    def __ldap_connection(self):
+        ldap.set_option(ldap.OPT_REFERRALS, 0)
+        conn = ldap.initialize(self.default_settings['SERVER_URI'])
+        conn.protocol_version = ldap.VERSION3
+        configure_tls(self.is_tls, os.environ.get(CERT_ENV_KEY, ''), conn)
+        conn.simple_bind_s(self.default_settings['BIND_DN'], self.default_settings['BIND_PASSWORD'])
+        try:
+            yield conn
+        finally:
+            conn.unbind_s()
+
+    def __get_groups_by_group_search(self):
+        with self.__ldap_connection() as conn:
+            groups = self.default_settings['GROUP_SEARCH'].execute(conn)
+        log.debug('Found %s groups: %s', len(groups), [i[0] for i in groups])
+        return groups
+
     def __create_rbac_groups(self, user):
         ldap_groups = list(zip(user.ldap_user.group_names, user.ldap_user.group_dns))
 
         for group in user.groups.filter(name__in=[i[0] for i in ldap_groups]):
-            ldap_group_dn = self.__det_ldap_group_dn(group.name, ldap_groups)
+            ldap_group_dn = self.__get_ldap_group_dn(group.name, ldap_groups)
             rbac_group = self.__get_rbac_group(group, ldap_group_dn)
             group.user_set.remove(user)
             rbac_group.group_ptr.user_set.add(user.user_ptr)
             if group.user_set.count() == 0:
                 group.delete()
 
-    @staticmethod
-    def __check_user(username: str):
+    def __check_user(self, ldap_user):
+        user_dn = ldap_user.dn
+        username = ldap_user._username  # pylint: disable=protected-access
+
         if User.objects.filter(username__iexact=username, type=OriginType.Local).exists():
             log.exception('usernames collision: `%s`', username)
-            raise AdcmEx('LDAP_USERNAMES_COLLISION')
+            return False
+
+        group_member_attr = self.default_settings['GROUP_TYPE'].member_attr
+        for _, group_attrs in self.__get_groups_by_group_search():
+            if user_dn.lower() in [i.lower() for i in group_attrs.get(group_member_attr, [])]:
```

Comment:
group_attrs.get(group_member_attr, []) will return empty list if there is no group_member_attr key in group_attrs but list don't have `lower` method